src/main/java/pl/cleankod/ApplicationInitializer.java

```diff
@@ -34,18 +46,42 @@ AccountRepository accountRepository() {
     }
 
     @Bean
-    ExchangeRatesNbpClient exchangeRatesNbpClient(Environment environment) {
-        String nbpApiBaseUrl = environment.getRequiredProperty("provider.nbp-api.base-url");
-        return Feign.builder()
+    ExchangeRatesNbpClient exchangeRatesNbpClient(Environment environment, CircuitBreakerRegistry circuitBreakerRegistry, ErrorDecoder errorDecoder) {
```

Comment:
do we really need to register 2 extra beans just to use them here? Maybe a simple method call would be sufficient?